mokapot/tabular_data.py

```diff
@@ -0,0 +1,603 @@
+    ) -> "TabularDataReader":
+        # Currently, we look only at the suffix, however, in the future we could
+        # also look into the file itself (is it ascii? does it have some "magic
+        # bytes"? ...)
+        suffix = file_name.suffix
+        if suffix in CSV_SUFFIXES:
+            reader = CSVFileReader(file_name, **kwargs)
+        elif suffix in PARQUET_SUFFIXES:
+            reader = ParquetFileReader(file_name, **kwargs)
+        else:
+            # Fallback
+            warnings.warn(
+                f"Suffix '{suffix}' not recognized in file name '{file_name}'."
+                " Falling back to CSV..."
+            )
+            reader = CSVFileReader(file_name, **kwargs)
+
+        if column_map is not None:
+            reader = ColumnMappedReader(reader, column_map)
+
+        return reader
+
+
+@typechecked
+class ColumnMappedReader(TabularDataReader):
+    def __init__(self, reader: TabularDataReader, column_map: dict[str, str]):
+        self.reader = reader
+        self.column_map = column_map
+
+    def get_column_names(self) -> list[str]:
+        return [
+            self.column_map.get(column, column)
+            for column in self.reader.get_column_names()
+        ]
+
+    def get_column_types(self) -> list:
+        return self.reader.get_column_types()
+
+    def _get_orig_columns(self, columns: list[str] | None) -> list[str] | None:
+        if columns is None:
+            return None
+
+        all_orig_columns = self.reader.get_column_names()
+        all_columns = self.get_column_names()
+        reverse_column_map = dict(zip(all_columns, all_orig_columns))
+        orig_columns = [reverse_column_map[column] for column in columns]
+        return orig_columns
+
+    def _get_mapped_dataframe(self, df: pd.DataFrame) -> pd.DataFrame:
+        # todo: enable this again...
+        # if self._returned_dataframe_is_mutable():
+        #     df.rename(columns=self.column_map, inplace=True, copy=False)
+        # else:
+        #     df = df.rename(columns=self.column_map, inplace=False, copy=False)
+        df = df.rename(columns=self.column_map, inplace=False)
+        return df
+
+    def read(self, columns: list[str] | None = None) -> pd.DataFrame:
+        df = self.reader.read(columns=self._get_orig_columns(columns))
+        return self._get_mapped_dataframe(df)
+
+    def get_chunked_data_iterator(
+        self, chunk_size: int, columns: list[str] | None = None
+    ) -> Generator[pd.DataFrame, None, None]:
+        orig_columns = self._get_orig_columns(columns)
+        for chunk in self.reader.get_chunked_data_iterator(
+            chunk_size, columns=orig_columns
+        ):
+            yield self._get_mapped_dataframe(chunk)
+
+
+def _types_from_dataframe(df: pd.DataFrame) -> list:
+    type_map = df.dtypes
+    column_names = df.columns.tolist()
+    return [type_map[column_name] for column_name in column_names]
```

Comment:
how is this different from `df.dtypes.tolist()`?

Reply:
It's not, I just wasn't aware of it ;) Removed the function completely and replaced calls to with calls to `.dtypes.tolist()`